### PR TITLE
Add a diagnostics extra (includes bokeh)

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/dask/dask/issues/4862

cc @philippegr (for awareness)
cc @mrocklin @TomAugspurger @quasiben @jrbourbeau @martindurant (for feedback)

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by jakirkham** [comment]
Planning on merging in 48hrs if no comments.

File: setup.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
   'dataframe': ['numpy >= 1.13.0', 'pandas >= 0.21.0', 'toolz >= 0.7.3',
                 'partd >= 0.3.10', 'cloudpickle >= 0.2.1'],
   'distributed': ['distributed >= 1.22'],
+  'diagnostics': ['bokeh >= 0.12.3'],
```
Comment (TomAugspurger) [inline setup.py]: In https://github.com/conda-forge/dask-feedstock/blob/master/recipe/meta.yaml, we require bokeh >=0.13.0

Comment (jakirkham) [reply setup.py]: Thanks Tom. I think I copied this from some reference in the codebase here. Will bump the version to match the feedstock.

Comment (jakirkham) [reply setup.py]: Updated. Please recheck.

**Comment by jcrist** [comment]
Can you add a test for this to `continuous_integration/travis/test_imports.sh`? This ensures that the packages that the extra adds support for are all importable with the given extra.

**Comment by jakirkham** [comment]
@jcrist, can I ask you to push the change you are looking for? I'm not entirely sure what you'd like to see. Also I've ran out of time to complete this for this cycle.

**Comment by jcrist** [comment]
Done

**Comment by jakirkham** [comment]
Thanks Jim. 🙂

**Comment by jakirkham** [comment]
Ack! Sorry for the noise. Botched something in git locally.

**Pull request merged**